src/components/taskDetails/index.tsx

```diff
@@ -0,0 +1,242 @@
+            const responseData = await updateTaskDetails(
+                state.editedDetails,
+                taskID
+            );
+            if (responseData.status === 204) {
+                initialDataRef.current = state.taskDetails;
+                toast(SUCCESS, 'Successfully saved');
+            }
+        } catch (err) {
+            toast(ERROR, 'Could not save changes');
+            dispatch({
+                type: 'setTaskDetails',
+                payload: initialDataRef.current,
+            });
+        }
+        dispatch({ type: 'reset', payload: initialDataRef.current });
+    }
+
+    function renderTextarea(name: string, value: string) {
+        return (
+            <textarea
+                className={classNames['textarea']}
+                name={name}
+                value={value}
+                data-testid="edit button"
+                onChange={(event) => handleChange(event)}
+            />
+        );
+    }
+
+    function renderButton(
+        buttonName: string,
+        clickHandler: (value: any) => void,
+        value?: boolean
+    ) {
+        if (isAuthorized) {
+            return (
+                <button
+                    type="button"
+                    className={classNames['button']}
+                    onClick={() => clickHandler(value)}
+                >
+                    {buttonName}
+                </button>
+            );
+        }
+    }
+
+    function renderLoadingComponent() {
+        if (isLoading) {
+            return <p className={classNames.textCenter}>Loading...</p>;
+        }
+        if (error) {
+            return (
+                <p className={classNames.textCenter}>Something went wrong!</p>
+            );
+        }
+    }
+
+    const shouldRenderParentContainer = () =>
+        !isLoading && !error && taskDetails;
+
+    return (
+        <>
+            <NavBar />
+            {renderLoadingComponent()}
+            {shouldRenderParentContainer() && (
+                <div className={classNames.parentContainer}>
+                    <div className={classNames.titleContainer}>
+                        {isEditing ? (
+                            renderTextarea('title', taskDetails?.title)
+                        ) : (
+                            <span
+                                data-testid="task-title"
+                                className={classNames.taskTitle}
+                            >
+                                {taskDetails?.title}
+                            </span>
+                        )}
+                        {!isEditing ? (
+                            renderButton('Edit', setIsEditing, true)
+                        ) : (
+                            <div className={classNames.editMode}>
+                                {renderButton('Cancel', onCancel)}
+                                {renderButton('Save', onSave)}
+                            </div>
+                        )}
+                    </div>
```

Comment:
Will it be cleaner to create an "editing" version of component separately?

Reply:
Eg - 

```js
isEditing ?
  <TaskDetailBodyEditMode />:
  <TaskDetailBodyViewMode />
```

Reply:
Updated with using dedicated components for Editing and Non-editing phases. ✅